xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1484,3 +1484,21 @@ bool DX::DeviceResources::SetMultithreadProtected(bool enabled) const
 
   return (wasEnabled == TRUE ? true : false);
 }
+
+bool DX::DeviceResources::IsGCNOrOlder() const
+{
+  if (CSysInfo::GetWindowsDeviceFamily() == CSysInfo::Xbox)
+    return false;
+
+  const DXGI_ADAPTER_DESC ad = GetAdapterDesc();
+
+  if (GetAdapterDesc().VendorId != PCIV_AMD)
+    return false;
+
+  const VideoDriverInfo driver = CWIN32Util::GetVideoDriverInfo(ad.VendorId, ad.Description);
+
+  if (CWIN32Util::IsDriverVersionAtLeast(driver.version, "31.0.22000.0"))
```

Comment:
Exceptionally driver info may be missing or corrupted in Windows Registry.

To avoid undefined behavior maybe check for valid in `VideoDriverInfo` struct before parse result:

```c++
  if (!driver.valid)
    return false;
```